packages/slate-react/src/components/editable.tsx

```diff
@@ -153,50 +153,56 @@ export const Editable = (props: EditableProps) => {
 
   // Whenever the editor updates, make sure the DOM selection state is in sync.
   useIsomorphicLayoutEffect(() => {
-    const { selection } = editor
-    const domSelection = window.getSelection()
-
-    if (state.isComposing || !domSelection || !ReactEditor.isFocused(editor)) {
-      return
-    }
+    setTimeout(() => {
```

Comment:
What's the need for this `setTimeout`?

Reply:
As the gif shows, in safari, when ever I use composition input, the selection will reset to the beginning of the `String` compnent (as the `span` tag). It seems in safari do reset in `layoutEffect` is not ok. 

Change to `useEffect` instend of `useLayoutEffect` will work fine too.

Also I added **reset text** to the `String` component, it will cause selection reset to the beginning too.

Reply:
If the *delay of reset selection* ok, I will change to `useEffect` instead of `setTimeout`

Reply:
`useEffect` sounds like a much better choice than `setTimeout` to me.